sequencer/src/state_signature.rs

```diff
@@ -38,47 +42,73 @@ pub(super) async fn state_signature_loop<N>(
         } = event
         {
             if let Some(leaf) = leaf_chain.first() {
-                let state = form_light_client_state(leaf, stake_table_comm);
-                let signature = context.sign_new_state(&state).await;
-                tracing::debug!(
-                    "New leaves decided. Latest block height: {}",
-                    leaf.get_height(),
-                );
+                match form_light_client_state(leaf, stake_table_comm) {
+                    Ok(state) => {
+                        let signature = context.sign_new_state(&state).await;
+                        tracing::debug!(
+                            "New leaves decided. Latest block height: {}",
+                            leaf.get_height(),
+                        );
 
-                if let Some(client) = client {
-                    let request_body = StateSignatureRequestBody {
-                        key: key.clone(),
-                        state,
-                        signature,
-                    };
-                    if let Err(error) = client
-                        .post::<()>("api/state")
-                        .body_binary(&request_body)
-                        .unwrap()
-                        .send()
-                        .await
-                    {
-                        tracing::warn!("Error posting signature to the relay server: {:?}", error);
+                        if let Some(client) = client {
+                            let request_body = StateSignatureRequestBody {
+                                key: key.clone(),
+                                state,
+                                signature,
+                            };
+                            if let Err(error) = client
+                                .post::<()>("api/state")
+                                .body_binary(&request_body)
+                                .unwrap()
+                                .send()
+                                .await
+                            {
+                                tracing::warn!(
+                                    "Error posting signature to the relay server: {:?}",
+                                    error
+                                );
+                            }
+                        }
                     }
+                    Err(err) => tracing::error!("Error generating light client state: {:?}", err),
                 }
             }
         }
     }
     tracing::info!("And now his watch has ended.");
 }
 
+fn hash_bytes_to_field(bytes: &[u8]) -> Result<CircuitField, PrimitivesError> {
+    // make sure that `mod_order` won't happen.
+    let bytes_len = ((<CircuitField as PrimeField>::MODULUS_BIT_SIZE + 7) / 8 - 1) as usize;
+    let elem = bytes
+        .chunks(bytes_len)
+        .map(CircuitField::from_le_bytes_mod_order)
+        .collect::<Vec<_>>();
+    Ok(VariableLengthRescueCRHF::<_, 1>::evaluate(elem)?[0])
+}
```

Comment:
are there reasons to use Rescue Hash here? why can't we just hash down to 32 bytes, and then use `jf_utils::bytes_to_field_elements`?

Reply:
No specific reason, We could use others as well

Reply:
yeah, let's try to minimize rescue usage wheneve possible, it's slow and not as time-tested. 
also try to reuse jf_utils, cuz converting bytes to fields is a common util. 
would you agree?